### PR TITLE
Fixes #22239: Windows policy generation

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/22239

### CODE REVIEW

**Comment by Fdall** [comment]
looks good

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: policies/rudderc/templates/technique.ps1.askama [hunk 1]
```diff
@@ -32,21 +34,25 @@ function {{ id }}
     TechniqueName = $techniqueName
     }
     {% match m.condition %}
-    {% when Some with (cond) %}
+    {%- when Some with (cond) %}
     $class = "{{ cond }}"
     if ($localContext.Evaluate($class)) {
         $methodParams = @{
-            {{ m.args }}
+        {% for arg in m.args %}
```
Comment (Fdall) [inline policies/rudderc/templates/technique.ps1.askama:41]: can you add 2 spaces at the start of the params in the hashmap?

File: policies/rudderc/tests/cases/general/reporting/technique.cf [hunk 2]
```diff
@@ -74,9 +74,9 @@ bundle agent call_b86ce2e5_d5b6_45cc_87e8_c11cca71d907(c_name, c_key, report_id,
     "b86ce2e5-d5b6-45cc-87e8-c11cca71d907_${report_data.directive_id}" usebundle => _method_reporting_context_v4("${c_name}", "${c_key}", "${report_id}");
     "b86ce2e5-d5b6-45cc-87e8-c11cca71d907_${report_data.directive_id}" usebundle => package_present("htop", "", "", ""),
                                              if => concat("debian");
-    "b86ce2e5-d5b6-45cc-87e8-c11cca71d907_${report_data.directive_id}" usebundle => _classes_noop(canonify("${class_prefix}_package_present_${name}")),
+    "b86ce2e5-d5b6-45cc-87e8-c11cca71d907_${report_data.directive_id}" usebundle => _classes_noop(canonify("${class_prefix}_package_present_htop")),
```
Comment (Fdall) [inline policies/rudderc/tests/cases/general/reporting/technique.cf:77]: `${class_prefix}` is not defined here

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Review by Fdall — APPROVED** [review]

**Pull request merged**